vcf.c

```diff
@@ -2650,9 +2732,8 @@ int bcf_fmt_array(kstring_t *s, int n, int type, void *data)
             { \
                 type_t v = convert(p); \
                 if ( is_vector_end ) break; \
-                if ( j ) kputc(',', s); \
-                if ( is_missing ) kputc('.', s); \
-                else e |= kprint < 0; \
+                if ( j ) e |= kputc_(',', s) < 0; \
+                e |= (is_missing ? kputc_('.', s) : kprint) < 0; \
```

Comment:
And we probably want `kputc()` here too, unless we ensure the kstring is properly terminated before the function returns.